### ISSUE TITLE
Integration with KOReader Possible?

### ISSUE THREAD

**Issue by MingLin-home** [clisboa/KoboMail#3]
Hi Clisboa,

This tool is so cool and useful! I really miss the send-to-kindle via email function in KOBO devices. Thanks for your great efforts!

As I use KOReader exclusively, I am wondering whether it is possible to keep KoboMail auto-sync in the background when KOReader is running. The issue here is that KOReader will kill Nickel process during its start so any hook/trigger based on NickelMenu or NickelDBus might not work at all.

My curruent workaround is to exit the KOReader which will restart the Nickel automatically and then manually run KoboMail. Clearly this is cumbersome. Really wish a better solution here.

I am new to Kobo dev hence might not understand your scripts and KOReader starting script very well. Please enlighten me if I was wrong about how they work under the hood.

**Comment by clisboa** [comment]
Hi, NickelDbus and NickelMenu are optional, you shouldn't have a problem using the automatic method of execution. 
This method relies on UDEV rules which are activated every time wifi connection is up. AFAIK that's completely separate to Nickel processes.
So I would recommend to try the auto execution type in your kobomail config file:

[execution_type]
    type = "auto"

Let me know if it works.

If not, it seems to be possible to execute shell scripts in Koreader: https://github.com/koreader/koreader/wiki/Shell-scripts
You could try to navigate to /usr/local/kobomail/kobomail_launcher.sh

**Comment by MingLin-home** [comment]
Hi Clisboa,

Thank you for the quick reply and the instruction!

I went through the .sh files of KoboMail again and find that it is using udev rules to trigger the launcher. So no more concerns about the NickelMenu/Dbus issues! 

I did not know KOReader can excute shell script directly. Thank you for sharing the wiki link! I just found another way myself: start the terminal emulator and write whatever we want and the save the command as ShortCut. Yet the fully automatic udev trigger is much better.

Wish KOReader could integrate KoboMail to its official plugins in near future!
